File: 4101-maximum-product-of-three-elements-after-one-replacement/4101-maximum-product-of-three-elements-after-one-replacement.go
func maxProduct(nums []int) int64 {
    var mx1 int64
    var mx2 int64
    n := len(nums)

    for i := 0; i < n; i++ {
        var absNum int64
        if nums[i] < 0 {
            absNum = int64(-nums[i])
        } else {
            absNum = int64(nums[i])
        }

        if absNum > mx2 {
            if absNum > mx1 {
                mx2 = mx1
                mx1 = absNum
            } else {
                mx2 = absNum
            }
        }
    }
    return int64(mx1 * mx2 * 100000)
}

File: 4101-maximum-product-of-three-elements-after-one-replacement/4101-maximum-product-of-three-elements-after-one-replacement.c
long long maxProduct(int* nums, int numsSize) {
    typedef long long ll;
    ll max1 = 0;
    ll max2 = 0;

    for(int i = 0; i < numsSize; i++){
        ll absNum = nums[i] < 0 ? -nums[i] : nums[i];

        if (absNum > max2) {
            if (absNum > max1) {
                max2 = max1;
                max1 = absNum;
                } 
            else
                max2 = absNum;
            
        }
    }

    return (ll) max1 * max2 * 100000;
}